e2e/test/iothub/service/RegistryManagerE2ETests.cs

```diff
@@ -144,5 +146,149 @@ public async Task RegistryManager_Query_Works()
+
+                IList<string> moduleIdsOnDevice = modulesOnDevice
+                    .Select(module => module.Id)
+                    .ToList();
+
+                Assert.AreEqual(moduleCount, moduleIdsOnDevice.Count);
+                for (int i = 0; i < moduleCount; i++)
+                {
+                    Assert.IsTrue(moduleIdsOnDevice.Contains(testModuleIds[i]));
+                }
+            }
+            finally
+            {
+                await Cleanup(client, testDeviceId).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Test basic lifecycle of a Device Identity.
+        /// This test includes CRUD operations only.
+        /// </summary>
+        [LoggedTestMethod]
+        public async Task ModulesClient_IdentityLifecycle()
+        {
+            string testDeviceId = $"IdentityLifecycleDevice{Guid.NewGuid()}";
+            string testModuleId = $"IdentityLifecycleModule{Guid.NewGuid()}";
+
+            Module module = null;
+            RegistryManager client = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            try
+            {
+                // Create a device to house the module
+                Device device = await client.AddDeviceAsync(new Device(testDeviceId)).ConfigureAwait(false);
+
+                // Create a module on the device
+                Module createResponse = await client.AddModuleAsync(
+                    new Module(testDeviceId, testModuleId)).ConfigureAwait(false);
+
+                createResponse.DeviceId.Should().Be(testDeviceId);
+                createResponse.Id.Should().Be(testModuleId);
+
+                // Get device
+                // Get the device and compare ETag values (should remain unchanged);
+                Module getResponse = await client.GetModuleAsync(testDeviceId, testModuleId).ConfigureAwait(false);
+
+                getResponse.ETag.Should().BeEquivalentTo(createResponse.ETag, "ETag value should not have changed between create and get.");
+
+                // Update a module
+                string managedByValue = "SomeChangedValue";
+                getResponse.ManagedBy = managedByValue;
+
+                Module updateResponse = await client.UpdateModuleAsync(getResponse).ConfigureAwait(false);
+
+                updateResponse.ManagedBy.Should().Be(managedByValue, "Module should have changed its managedBy value");
+
+                // Delete the device
+                // Deleting the device happens in the finally block as cleanup.
+            }
+            finally
+            {
+                await Cleanup(client, testDeviceId).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Test basic operations of a module's twin.
+        /// </summary>
+        [LoggedTestMethod]
+        public async Task ModulesClient_DeviceTwinLifecycle()
```

Comment:
how is this different from identity lifecycle test above?

Reply:
This one is more about the module's twin. It tests updating the twin with a new property

Reply:
Since we can't create or delete twins, only the update gets tested

Reply:
since it's an update operation nevertheless, be it update twin metadata, or twin properties, can we not add it to the above test itself?